src/Views/Form/Templates/Classic/resources/js/not-jquery.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Collect all siblings until a selector.
+ *
+ * @param {Node} node
+ * @param {string} selector
+ *
+ * @return {Node[]}
+ */
+export function nextUntil(node, selector) {
+	const siblings = [];
+	const sibling = node.nextElementSibling;
+
+	while (sibling) {
+		if (sibling.matches(selector)) break;
+
+		siblings.push(sibling);
+		sibling = sibling.nextElementSibling;
+	}
+
+	return siblings;
+}
+
+/**
+ * The missing after version of insertBefore.
+ *
+ * @param {Node} newNode
+ * @param {Node} referenceNode
+ */
+export function insertAfter(newNode, referenceNode) {
+	referenceNode.parentNode.insertBefore(newNode, referenceNode.nextSibling);
+}
+
+/**
+ * @param {Node} node
+ */
+export function removeNode(nodeToRemove) {
+	nodeToRemove.parentNode.removeChild(nodeToRemove);
+}
+
+/**
+ * Creates a DOM Node from an HTML string.
+ *
+ * @param {string} htmlString
+ * @returns {Node}
+ */
+export function nodeFromString(htmlString) {
+	const temp = document.createElement('template');
+	htmlString.trim();
+	temp.innerHTML = htmlString;
+	return temp.content.firstChild;
+}
+
+/**
+ * Execute a function when the DOM is fully loaded.
+ *
+ * @param {function} fn
+ */
+export const domIsReady = fn => document.readyState !== 'loading'
```

Comment:
Is the `document.readyState` check really needed here? Why don't just add a listener to the `DOMContentLoaded` event?

Reply:
This function is ultimately derived from jQuery’s old `.ready()` function. jQuery does a little extra to support IE. Checking `document.readyState !== 'loading'` is for `async` scripts, since it is possible to miss the `DOMContentLoaded` event. `setTimeout(fn, 0)` makes the code execute asynchronously like how an event handler is called.

Now, I didn’t write this function with classic template in mind; I just copied and pasted this from an old PR to Sage that I had made to replace jQuery. Here we aren’t loading the JS asynchronously, but I would be tempted to just leave the extra check since this is WordPress and users can change how stuff is loaded.

I was going to bring this up at some point: WordPress has this function standalone as an include `wp-dom-ready` (I might have spelled something wrong). We can just enqueue that instead. Also, we could use jQuery just for this since it’s enqueued already. I’m personally for not using jQuery where it makes sense not to use it, so that there's less work later on when we drop it.

Reply:
I’m actually going to switch to follow this approach here. It potentially fires before the DOM is interactive, which we want since we are swapping out content.

https://gomakethings.com/the-quickest-way-to-detect-when-the-dom-is-ready/

Reply:
I don't mind if it's reliable.